bids/layout/layout.py

```diff
@@ -299,12 +299,30 @@ def __getattr__(self, key):
 
     def __repr__(self):
         """Provide a tidy summary of key properties."""
-        # TODO: Replace each nested list comprehension with a single DB query
-        n_sessions = len([session for isub in self.get_subjects()
-                          for session in self.get_sessions(subject=isub)])
-        n_runs = len([run for isub in self.get_subjects()
-                      for run in self.get_runs(subject=isub)])
-        n_subjects = len(self.get_subjects())
+        n_subjects = len(
+            [s.value
+             for s in self.session.query(Tag).filter_by(
+                 entity_name='subject').group_by(Tag._value)]
+            )
```

Comment:
Is this not just:

```Python
n_subjects = len(self.session.query(Tag).filter_by(entity_name='subject').group_by(Tag._value))

Reply:
Yes, this is a hold over good even when I was using the actual subject list for something